bin/util.py

```diff
@@ -748,9 +749,18 @@ def get_memory_limit(kwargs=None):
         kwargs.pop('memory')
     return memory_limit
 
+class ExecCode(Enum):
+    ACCEPTED = 1
+    REJECTED = 2
+    ERROR = 3
+    TIMEOUT = 4
+
+    def __bool__(self):
+        return self == ExecCode.ACCEPTED
 
 class ExecResult:
-    def __init__(self, ok, duration, timeout_expired, err, out, verdict=None, print_verdict=None):
+    def __init__(self, code, ok, duration, timeout_expired, err, out, verdict=None, print_verdict=None):
+        self.code = code
```

Comment:
Add some comments here.
- `code` is the return code (or None) (maybe rename to `returncode`, similar to `process.returncode`?)
- `ok` is the `ExecCode` above, but would be clearer to rename this to `status` or so also probably?

Reply:
can you easily refactor this? ^^' (there are other things that are called `.ok` and i didn't want to break them...)

Reply:
It feels kind of weird to have the `ExecResult.ok` (to be renamed to `ExecCode.status` field have two semantics: one as a "boolean-able" and one as an actual enumeration :thinking: If we're rethinking the terminology here anyway, I would propose that we could separate the two? For example:
```py
class ExecResult:
    def __init__(self, returncode, status, duration, ...):
        self.returncode = returncode
        self.status = status
        ...
    
    def is_ok(self):
        self.status == ExecStatus.ACCEPTED
```
Then we also don't need the coercion method `ExecStatus.__bool__` anymore :slightly_smiling_face:

Reply:
I don't mind the implicit conversion to bool so much though; it makes call sites nice and simple. But ok with either.

Reply:
After renaming .ok to .status I feel like maybe this is sightly more appropriate now. as you wish.

Reply:
I think I'm fine with the coercion to boolean if we check in the constructor of `ExecResult` that `status` really is one of the enum values, and not something silly like `None` :slightly_smiling_face: (note that the contructor currently still uses `ok` instead of `status` :stuck_out_tongue:

Reply:
Oh I did that but forgot to save the file :)

Checking type is a good idea indeed.

Reply:
grrr;`pytest` is a bit broken for me:
```
FATAL ERROR: Did not find problem.yaml. Are you running this from a problem directory?
/usr/lib/python3.11/site-packages/_pytest/main.py:306: PluggyTeardownRaisedWarning: A plugin raised an exception during an old-style hookwrapper teardown.
Plugin: terminalreporter, Hook: pytest_sessionfinish
SystemExit: 1
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config.hook.pytest_sessionfinish(
```

not sure what's wrong, and CI is having some different weird issue now...